angular/src/components/export-scope-callout.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, OnInit } from "@angular/core";
+
+import { OrganizationService } from "jslib-common/abstractions/organization.service";
+import { StateService } from "jslib-common/abstractions/state.service";
+
+@Component({
+  selector: "app-export-scope-callout",
+  templateUrl: "export-scope-callout.component.html",
+})
+export class ExportScopeCalloutComponent implements OnInit {
+  @Input() organizationId: string = null;
+
+  show: boolean = false;
+  scopeConfig: {
+    title: string;
+    description: string;
+    scopeIdentifier: string;
+  };
+
+  constructor(
+    protected organizationService: OrganizationService,
+    protected stateService: StateService
+  ) {}
+
+  async ngOnInit(): Promise<void> {
+    if (!(await this.organizationService.hasOrganizations())) {
+      return;
+    }
+    this.show = true;
```

Comment:
Suggestion: move this line to the end of this method after all the async calls, so that text (e.g. org name or user's email) isn't popping in.

Reply:
I moved this as requested, but I don't _think_ the situation described here is possible? ngOnInit is [supposed](https://angular.io/api/core/OnInit#ngoninit) to run completely before any content is loaded.

Regardless, it's moved, so just trying to see if you know something I don't about this behavior.

Reply:
@differsthecat The providers tab was exactly what I'm thinking of. But @addisonbeck I think the simplest answer is that Angular calls the lifecycle hooks at the appropriate times, but the DOM rendering does not wait for them to return/resolve. This is why we have spinners or `*ngIf` directives. Here's a simple-as-possible reproduction of this behaviour: https://stackblitz.com/edit/angular-4ariyn